tests/acceptance/test_sasl.py

```diff
@@ -0,0 +1,69 @@
+# MIT License
+# Copyright (c) 2020-2024 Pau Freixes
+
+import asyncio
+import logging
+import os
+import sys
+from pathlib import Path
+
+import pytest
+
+from emcache import MemcachedHostAddress, create_client
+from emcache.client_errors import AuthenticationError
+
+pytestmark = pytest.mark.asyncio
+
+logging.basicConfig(level=logging.DEBUG)
+
+with open(Path(os.path.dirname(__file__)) / "data" / "auth_pwd.txt", "r") as f:
+    username, password = f.readline().rstrip().split(":")
+
+
+@pytest.fixture()
+async def authed_client(event_loop):
+    client = await create_client(
+        [MemcachedHostAddress("localhost", 11214)], timeout=1.0, username=username, password=password
+    )
+    try:
+        yield client
+    finally:
+        await client.close()
+
+
+class TestAuth:
+    @pytest.mark.skipif(sys.platform == "darwin", reason="This server is not built with SASL support.")
+    async def test_auth(self, authed_client):
+        await authed_client.get(b"key")
+
+    @pytest.mark.skipif(sys.platform == "darwin", reason="This server is not built with SASL support.")
+    async def test_auth_error_no_userpass(self):
+        with pytest.raises(asyncio.TimeoutError):
```

Comment:
Why `TimeoutError`?  shouldn't this be returning an auth exception?

Reply:
I think we need to add error handling for the fetch method.
https://github.com/emcache/emcache/issues/162

Reply:
Oks, did not know this. Good catch. Having the feeling that #162 could be a blocker for having this one deployed, depending on the effort for addressing the #162 

How easy would be to change the multiline parser? and moreover could be other commands impacted for this issue?

Reply:
I am ready to create a new PR on this issue

Reply:
And so what I want to do. Add a check inside AsciiMultiLineParser. When receiving an error message from memcached, we simply return a message without \n\r. In a good case, we parse using keys, values, flags, cas.
Then you need to check after receiving the result like all other commands.

And lastly, we change tests with timeout to CommandError errors

Reply:
```python
result = await self._fetch_command(b"get", key)
if len(result) == 1:
    raise CommandError(f"Command finished with error, response returned {result}")

keys, values, flags, _ = result
```

Reply:
@pfreixes , Is this expected behavior?

Reply:
I think there must be a command error here, since the user is not even trying to connect, or do you still insist on autherror?